Assets/Mirror/Components/NetworkSceneChecker.cs

```diff
@@ -0,0 +1,119 @@
+using UnityEngine;
+using System.Collections.Generic;
+using UnityEngine.SceneManagement;
+
+namespace Mirror
+{
+    /// <summary>
+    /// Component that controls visibility of networked objects between scenes.
+    /// <para>Any object with this component on it will only be visible to other objects in the same scene</para>
+    /// <para>This would be used when the server has multiple additive subscenes loaded to isolate players to their respective subscenes</para>
+    /// </summary>
+    [AddComponentMenu("Network/NetworkSceneChecker")]
+    [RequireComponent(typeof(NetworkIdentity))]
+    [HelpURL("https://mirror-networking.com/docs/Components/NetworkSceneChecker.html")]
+    public class NetworkSceneChecker : NetworkBehaviour
+    {
+        /// <summary>
+        /// Flag to force this object to be hidden from all observers.
+        /// <para>If this object is a player object, it will not be hidden for that client.</para>
+        /// </summary>
+        [Tooltip("Enable to force this object to be hidden from all observers.")]
+        public bool forceHidden;
+
+        static Dictionary<string, HashSet<NetworkIdentity>> sceneCheckerObjects;
+
+        [Scene, SerializeField]
+        string currentScene;
+
+        [ServerCallback]
+        void Awake()
+        {
+            if (sceneCheckerObjects == null)
+                sceneCheckerObjects = new Dictionary<string, HashSet<NetworkIdentity>>();
```

Comment:
you can create this by default. it's static anyway, not worth the extra code

Reply:
Calling the constructor 100 times is safe / acceptable?

Reply:
if you initialize a static field,  it gets called only once.

Reply:
Done